support/ci/cargo_update.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+set -euo pipefail 
+ 
+if ! "${BUILDKITE:-false}"; then 
+  echo "This script does not appear to be running in Buildkite. Exiting!" 
+  exit 1
+fi
+
+# shellcheck source=./support/ci/shared.sh 
+source ./support/ci/shared.sh 
+
+branch="ci/cargo-update-$(date +"%Y%m%d%H%M%S")"
+git checkout -b "$branch"
+
+toolchain="$(get_toolchain)"
+
+install_rustup
+install_rust_toolchain "$toolchain"
+
+echo "--- :box: Cargo Update"
+cargo +"$toolchain" update
+echo "--- :box: Cargo Check"
+cargo +"$toolchain" check --quiet --all --tests
+
+git add Cargo.lock
+
+
+git commit -s -m "Update Cargo.lock"
+
+# https://expeditor.chef.io/docs/reference/script/#open-pull-request
+open_pull_request
+
+git checkout master 
+git branch -D "$branch"
```

Comment:
I don't think this is what you're looking for if you want to delete the remove branch. IIRC this only deletes the local branch. I believe you need a git push with --delete (or `:` if you're feelin nasty) as well?

Reply:
That would end up closing the PR, though.

This is good hygiene, but not 100% necessary if we're on transient instances.

Reply:
As long as the PR gets merged and the delete_on_merge function hppens that's cool!